robot/impl/local_robot.go

```diff
@@ -225,6 +225,55 @@ func (r *localRobot) ModuleAddress() (string, error) {
 	return r.webSvc.ModuleAddress(), nil
 }
 
+// RemoveOrphanedResources is called by the module manager when a module has
+// changed (either due to crash or successful restart) and some or all of the
+// resources it handled have been orphaned. It will remove the orphaned
+// resources from the resource tree and Close them.
+func (r *localRobot) RemoveOrphanedResources(ctx context.Context,
```

Comment:
I don't love this solution of MM -> LR -> RM to handle orphaned resources. On top of that, I honestly don't fully understand what it does: the code of this function is frankensteined from `resourceManager.markRemoved` and `localRobot.Reconfigure`.